### PR TITLE
Feature: Add read receipt toggle in group details screen

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Create a new cell with to UISwitch for toggling read receipts setting of the conversation.

## Dependencies

Needs releases with:

- [ ] Data model from `feature/read-receipts`

### CODE REVIEW

**Comment by johnxnguyen** [comment]
Isn't the read receipt option not available in non teams? The new cell appears in both snapshots.

**Comment by johnxnguyen** [comment]
It would be nice to have a little bit of a description about the motivation to separate the `GroupDetailsOptionCell` into two protocols.

File: Wire-iOS/Sources/UserInterface/Components/Views/DetailsCollectionViewCell.swift [hunk 1]
```diff
@@ -21,7 +21,21 @@ import UIKit
 class DetailsCollectionViewCell: SeparatorCollectionViewCell {
 
     private let leftIconView = UIImageView()
-    private let accessoryIconView = UIImageView()
+    public var rightAccessoryView: UIView? {
+        didSet {
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/Components/Views/DetailsCollectionViewCell.swift]: Why did this change? It creates a lot of casting into `UIImageView`.

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Views/DetailsCollectionViewCell.swift]: Yes, it would be better to create two kinds of cell base classes.

**Comment by johnxnguyen** [comment]
I'm working on the receipt option cell in the group creation flow. Visually it looks identical to the one in the group details, but it's different in that it's not actually tied to a conversation. Do you think we could refactor out the visual part of the cells, then subclass/implement a protocol (I think this was part of your original idea to separate the protocols?) to then configure that cell for the conversation? Otherwise we will end up implementing (almost) the same thing twice in two different places.

File: Wire-iOS/Sources/UserInterface/GroupDetails/Cells/GroupDetailsReceiptOptionsCell.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireExtensionComponents
+
+class GroupDetailsReceiptOptionsCell: DetailsCollectionViewCell & ConversationOptionsConfigurable {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/GroupDetails/Cells/GroupDetailsReceiptOptionsCell.swift]: @johnxnguyen When you need a cell with UISwitch but not link to ZMConversation, you can use GroupDetailsReceiptOptionsCell. I will move ConversationOptionsConfigurable to extension.

**Pull request merged**